pkg/node/node.go

```diff
@@ -145,6 +149,42 @@ func getStagingPath(logger *logrus.Entry, stagingPath string) string {
 	return stagingPath
 }
 
+func (s *CSINodeService) processFakeAttachInNodeStageVolume(ll *logrus.Entry, volumeCR *volumecrd.Volume, targetPath string, isFakeAttachNeed bool) error {
+	volumeID := volumeCR.Spec.Id
+	if isFakeAttachNeed {
+		if volumeCR.Annotations[fakeAttachVolumeAnnotation] != fakeAttachVolumeKey {
+			volumeCR.Annotations[fakeAttachVolumeAnnotation] = fakeAttachVolumeKey
+			ll.Warningf("Adding fake-attach annotation to the volume with ID %s", volumeID)
+			s.VolumeManager.recorder.Eventf(volumeCR, eventing.FakeAttachInvolved,
+				"Fake-attach involved for volume with ID %s", volumeID)
+		}
+		// mount fake device in the non-fs mode
+		if volumeCR.Spec.Mode != apiV1.ModeFS {
+			fakeDevice, err := s.VolumeManager.createFakeDeviceIfNecessary(ll, volumeCR)
+			if err != nil {
+				ll.Errorf("unable to create fake device in stage volume request with error: %v", err)
+				return status.Error(codes.Internal, fmt.Sprintf("failed to create fake device in stage volume: %s", err.Error()))
+			}
+			volumeCR.Annotations[fakeDeviceVolumeAnnotation] = fakeDevice
+			if err := s.fsOps.PrepareAndPerformMount(fakeDevice, targetPath, true, false); err != nil {
+				ll.Errorf("unable to mount fake device in stage volume request with error: %v", err)
+				return status.Error(codes.Internal, fmt.Sprintf("failed to mount device in stage volume: %s", err.Error()))
+			}
+		}
+	} else if volumeCR.Annotations[fakeAttachVolumeAnnotation] == fakeAttachVolumeKey {
```

Comment:
currently,  is fake attach only need work in node stage phase?  how about node unstage phase?